src/ddscxx/include/dds/sub/detail/SampleRef.hpp

```diff
@@ -0,0 +1,117 @@
+#ifndef OMG_DDS_PUB_DETAIL_SAMPLEREF_HPP_
+#define OMG_DDS_PUB_DETAIL_SAMPLEREF_HPP_
+
+/* Copyright 2010, Object Management Group, Inc.
+ * Copyright 2010, PrismTech, Corp.
+ * Copyright 2010, Real-Time Innovations, Inc.
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+namespace dds
+{
+namespace sub
+{
+namespace detail
+{
+template <typename T>
+class SampleRef;
+}}}
+
+#include <dds/sub/SampleInfo.hpp>
+#include <org/eclipse/cyclonedds/topic/datatopic.hpp>
+
+namespace dds
+{
+namespace sub
+{
+namespace detail
+{
+template <typename T>
+class SampleRef
+{
+public:
+    SampleRef()
+    {
+      this->data_ = nullptr;
+    }
+
+    SampleRef(ddscxx_serdata<T>* d, const dds::sub::SampleInfo& i)
+    {
+        this->data_ = d;
+        this->info_ = i;
+    }
+
+    SampleRef(const SampleRef& other)
+    {
+        copy(other);
+    }
+
+    virtual ~SampleRef()
+    {
+        ddsi_serdata_unref(data_);
+    }
+
+    SampleRef& operator=(const SampleRef& other)
+    {
+        return copy(other);
+    }
+
+    SampleRef& copy(const SampleRef& other)
+    {
+        this->data_ = other.data_;
```

Comment:
Shouldn't this do `ddsi_serdata_ref(other.data_)`?

Reply:
You are correct. I will correct it.